### PR TITLE
Refactor: Handling Unrecognized Arguments for Prepare

### PR DESCRIPTION
**Pull request by selmanozleyen**
These are the things I want to do first.  related: #596

From reading through ottjax only the `__call__` functions on solvers absorb kwargs that aren't recognized. And they are because either they are given to create_geometry methods or create_initializer methods:

- https://github.com/ott-jax/ott/blob/458e265a1282d417f87bc0e07a5b6b77ffe9ae50/src/ott/initializers/quadratic/initializers.py#L134-L155
- https://github.com/ott-jax/ott/blob/458e265a1282d417f87bc0e07a5b6b77ffe9ae50/src/ott/initializers/linear/initializers_lr.py#L232-L243

So this is kind of good news because from what I read from moscot, call_fn arguments are filtered before here:
https://github.com/theislab/moscot/blob/e21677e6fbf1e5484b1c69a8ba55587dbf40f009/src/moscot/base/solver.py#L145-L164

Overall this means:
- We can fix the`prepare` methods absorbing kwargs ourselves in moscot, by simply removing the usage of kwargs in concrete classes, so I will dedicate this PR to it
- But for `solve` we will need to refactor things from ottjax. It won't be too hard and I already spotted where they are being ignored. If we refactor ottjax we might not need any changes in moscot since the error will be thrown from there.

Related issues (will be closed after merge):
- https://github.com/theislab/moscot/issues/698
- https://github.com/theislab/moscot/issues/699
- https://github.com/theislab/moscot/issues/700
- https://github.com/theislab/moscot/issues/701

I separated some issues from this PR so that we can work on them later:
- https://github.com/theislab/moscot/issues/716 : I did this one because when inheriting docstring GW removed the arguments for linear, and this lead to duplicate docstrings
- https://github.com/theislab/moscot/issues/715 : Because although in terms of software, current PR makes sense, there might be some cases where a parent classes argument is ignored/fixed but is not yet discovered. I am not confident about this as I am not completely familiar with the Biology use cases of all classes.
- https://github.com/theislab/moscot/issues/721

Now that the tests pass these are the things to do:
- [x] clean up code (most likely requires refactoring of `handle_joint_attr`)
- [x] update docstrings
- [x] clean the TODO's and ensure that everyone agrees on the new docs and docstrings
- [x] write tests to ensure it fails (not sure if tests are necessary at this point since the checks are static now, will probably figure out cases to test when working on https://github.com/theislab/moscot/issues/715)

### CODE REVIEW

File: src/moscot/problems/space/_alignment.py [hunk 1]
```diff
@@ -128,23 +132,56 @@ def prepare(
...
+                raise ValueError("Cannot normalize spatial coordinates when `x_callback` and `y_callback` are set.")
+                warnings.warn(
+                    "Ignoring `normalize_spatial` as `x_callback` and `y_callback` are set.",
+                    UserWarning,
+                    stacklevel=2,  # TODO: check these later
+                )
+        # TODO: what if one is spatial norm?
+        if x_callback == "spatial-norm" and y_callback == "spatial-norm":
+            x = y = {}
+        xy, xy_callback, xy_callback_kwargs = handle_joint_attr(joint_attr, xy_callback, xy_callback_kwargs)
```
Comment (giovp) [inline src/moscot/problems/space/_alignment.py:156]: > `# TODO: what if one is spatial norm?`

it must be in either both or none, as there is no use case for only one, hence they are set both by `normalize_spatial` arg

also this apply only for the alignment, as in the mapping it can be only in one

**Review by giovp — APPROVED** [review]
fantastic work @selmanozleyen !

File: src/moscot/problems/space/_alignment.py (hunk 1 above)
Comment (selmanozleyen) [reply src/moscot/problems/space/_alignment.py:156]: so should I throw an error when only one is given?

Comment (MUCDK) [reply src/moscot/problems/space/_alignment.py:156]: also for spatiotemporal. 
I think a warning would be good

**Review by MUCDK — APPROVED** [review]
Really great work Selman! Thanks so much!

File: src/moscot/base/problems/birth_death.py [hunk 2]
```diff
@@ -159,14 +163,53 @@ class BirthDeathProblem(BirthDeathMixin, OTProblem):
         Keyword arguments for :class:`~moscot.base.problems.OTProblem`.
     """  # noqa: D205
 
+    def prepare(
```
Comment (MUCDK) [inline src/moscot/base/problems/birth_death.py:166]: remove the prepare method and pass it via `marginal_kwargs`

**Comment by selmanozleyen** [comment]
@MUCDK  the problem is even though 3.10 passes 3.9 now fails. The results change between these versions

**Comment by MUCDK** [comment]
Ok can we skip the test for python 3.9?

**Comment by selmanozleyen** [comment]
Sure will update the tests in a new PR then
